templates/deployment_backend.yaml

```diff
@@ -132,6 +132,13 @@ spec:
           - name: "{{ $key }}"
             value: "{{ $value }}"
           {{- end }}
+          {{- range .Values.environmentSecrets }}
+          - name: {{ .name }}
+            valueFrom:
+              secretKeyRef:
+                name: {{ .secretKeyRef.name }}
+                key: {{ .secretKeyRef.key }}
+          {{- end }}
```

Comment:
I typically see this handled with just a singmple `toYaml`:

```
          env:
            {{- toYaml .Values.application.environment | nindent 12 }}
```

It would be a backwards-incompatible change, but then this chart would instantly handle all the current and future complexity of env var descriptions. Not sure how the maintainers would feel about that, though. I want this feature far more than I want this feature to be perfect. :)

Reply:
yeaa I agree the single `toYaml` is much nicer, but we need to think more about breaking changes 😬  let's get your changes landed first!